content/documents/configuring-user-application-data-access.mdx

```diff
@@ -0,0 +1,502 @@
+SecurityProfiles.Create({
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": [
+    "Shopper"
+  ]
+})
+.then((securityProfile) => {
+    console.log(securityProfile);
+})
+.catch((ex) => console.log(ex));
+`,
+typescript: `import { SecurityProfiles, SecurityProfile, OrderCloudError } from "ordercloud-javascript-sdk";\n
+SecurityProfiles.Create({
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": [
+    "Shopper"
+  ]
+})
+.then((securityProfile: SecurityProfile) => {
+  console.log(securityProfile);
+})
+.catch((ex: OrderCloudError) => console.log(ex));`,
+csharp: `using OrderCloud.SDK; \n
+var data = new SecurityProfile
+{
+  "ID": "defaultbuyer",
+  "Name": "Default Buyer",
+  "Roles": new ApiRole[] { ApiRole.Shopper }
+};
+try {
+  SecurityProfile response = await SecurityProfiles.Create(data);
+  Console.WriteLine(response)
+} catch(OrderCloudException ex) {
+  Console.WriteLine(ex.Message);
+}
+`
+}}
+/>
+
+### Create a seller security profile
+Next, we can create a security profile for our seller users.  Let's create a security profile, name it "Default Seller" with a self-descriptive ID, and specify the `FullAccess` role in the Roles array, which gives our sellers access to any resource in the API.  As a best practice, you would want to control the roles on a more granular level and only assign roles necessary for the user as a security precaution, but for our example, `FullAccess` will work.  You can find more details on security profiles [here](knowledge-base/security-profiles).
+<CodeExample
+  content={{
+    http: `POST https://sandboxapi.ordercloud.io/v1/securityprofiles HTTP/1.1
+Authorization: Bearer INSERT_ACCESS_TOKEN_HERE
+Content-Type: application/json; charset=UTF-8\n
+{
+  "ID": "defaultseller",
+  "Name": "Default Seller",
+  "Roles": [
+    "FullAccess"
+  ]
+}`,
+javascript: `import { SecurityProfiles } from "ordercloud-javascript-sdk";\n
+SecurityProfiles.Create({
+  "ID": "defaultseller",
+  "Name": "Default Seller",
+  "Roles": [
+    "FullAccess"
+  ]
+})
+.then((securityProfile) => {
+    console.log(securityProfile);
+})
+.catch((ex) => console.log(ex));
+`,
+typescript: `import { SecurityProfiles, SecurityProfile, OrderCloudError } from "ordercloud-javascript-sdk";\n
+SecurityProfiles.Create({
+  "ID": "defaultseller",
+  "Name": "Default Seller",
+  "Roles": [
+    "FullAccess"
+  ]
+})
+.then((securityProfile: SecurityProfile) => {
+  console.log(securityProfile);
+})
+.catch((ex: OrderCloudError) => console.log(ex));`,
+csharp: `using OrderCloud.SDK; \n
+var data = new SecurityProfile
+{
+  "ID": "defaultseller",
+  "Name": "Default Seller",
+  "Roles": new ApiRole[] { ApiRole.FullAccess }
+};
+try {
+  SecurityProfile response = await SecurityProfiles.Create(data);
+  Console.WriteLine(response)
+} catch(OrderCloudException ex) {
+  Console.WriteLine(ex.Message);
+}
+`
+}}
+/>
+
+### Assign security profiles to users
```

Comment:
This section is incorrect. Your example is assigning the security profile to the buyer **company** of id `buyeruser`. In general we do not want to encourage user specific assignments. So reword this so that it reads "let's assign the buyer profile to the entire buyer company so that any buyer user within it will inherit the same security profile".

Reply:
Oh, good catch there, that was a typo and should have said `UserID: buyeruser`, but I will update the verbiage and example here to assign the security profile to the entire buyer company.  I was not aware that this was the recommended method.

Reply:
In general we encourage user group as the most specific assignment party. I thought that user specific assignments weren't a thing anymore, but perhaps that's only related to product assignments... https://docs.ordercloud-qa.com/api-reference/product-catalogs/products/save-assignment

Not sure if user specific SP assignments are actually discouraged by platform team or not.